patches/api/0455-itemstack-expose-tooltip-lines.patch

```diff
@@ -0,0 +1,87 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Yannick Lamprecht <yannicklamprecht@live.de>
+Date: Mon, 22 Jan 2024 13:27:18 +0100
+Subject: [PATCH] itemstack expose tooltip lines
+
+
+diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
+index a4b38f284d4fea7df7f9df9bf44e4f68fefaf20f..3f2aca1b096a36d5a15ecee50b8bc107ab173416 100644
+--- a/src/main/java/org/bukkit/UnsafeValues.java
++++ b/src/main/java/org/bukkit/UnsafeValues.java
+@@ -260,4 +260,5 @@ public interface UnsafeValues {
+ 
+     String getStatisticCriteriaKey(@NotNull org.bukkit.Statistic statistic);
+     // Paper end
++    @NotNull java.util.List<net.kyori.adventure.text.Component> tooltipLines(@NotNull ItemStack itemStack, @Nullable org.bukkit.entity.Player player, @NotNull org.bukkit.inventory.TooltipOption options); // Paper - expose itemstack tooltip lines
+ }
+diff --git a/src/main/java/org/bukkit/inventory/ItemStack.java b/src/main/java/org/bukkit/inventory/ItemStack.java
+index 36e3fbc727cd748aa138f52976154ba32954cd87..ed3cf73e8ace3cf8ed9c0046827d0f986bb58f1d 100644
+--- a/src/main/java/org/bukkit/inventory/ItemStack.java
++++ b/src/main/java/org/bukkit/inventory/ItemStack.java
+@@ -1005,4 +1005,17 @@ public class ItemStack implements Cloneable, ConfigurationSerializable, Translat
+         return type.isAir() || amount <= 0;
+     }
+     // Paper end
++    // Paper start - expose itemstack tooltip lines
++    /**
++     * Returns the tooltip lines of the Itemstack.
++     * Passing the player maps additional data that are player
++     * or player world specific.
++     *
++     * @param player the player is optional, but triggers additional mapping.
++     * @param tooltipOption these options describe as whom the tooltips should be viewed
++     * @return a list of components or a empty list if non-available
++     */
++    public @NotNull List<net.kyori.adventure.text.Component> tooltipLines(@Nullable org.bukkit.entity.Player player, @NotNull TooltipOption tooltipOption) {
++        return Bukkit.getUnsafe().tooltipLines(this, player, tooltipOption);
++    } // Paper end - expose itemstack tooltip lines
+ }
+diff --git a/src/main/java/org/bukkit/inventory/TooltipOption.java b/src/main/java/org/bukkit/inventory/TooltipOption.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..513ab14c5d8c2b22fe6c8c0c0a95d1af5ba2144b
+--- /dev/null
++++ b/src/main/java/org/bukkit/inventory/TooltipOption.java
```

Comment:
I think this should go in the paper package

Reply:
Addressed.